### PR TITLE
fix(index item tags): hide icons from a11y tree; add descriptive aria labels

### PR DESCRIPTION
**Pull request by dakahn**
ref #1744 

In my screen reader testing the index tags were being read twice -- once for the list item itself and another time for the image (icon) inside of it. It was also difficult to discern the information the tags were trying to convey. This PR does two things:

- hides the icons from the accessibility tree with `aria-hidden`
- adds more descriptive aria labeling so the information (framework type, design asset type, and maintainer) is more understandable
- BONUS THING: removes the word 'image' in our asset placeholder which was being flagged by jsx-a11y as redundant and unnecessary given it's context in the accessibility tree (cool!)

### note for testing
The elements are still read as buttons which is confusing. I started a more thorough discussion of this at the referenced issue above 🏄

### CODE REVIEW

File: src/components/ComponentIndexPage/ComponentIndexListItem.js [hunk 1]
```diff
@@ -61,10 +62,7 @@ const ComponentIndexListItem = React.memo(
       );
     } else {
       img = (
-        <img
-          src={placeholder}
-          alt={`Placeholder image for the ${name} component`}
-        />
+        <img src={placeholder} alt={`Placeholder for the ${name} component`} />
```
Comment (emyarod) [inline src/components/ComponentIndexPage/ComponentIndexListItem.js:65]: just wondering is this formatting change intentional or was it based on personal settings?

Comment (dakahn) [reply src/components/ComponentIndexPage/ComponentIndexListItem.js:65]: I should just be picking up _our_ prettier config, unless something changed with VS Code/prettier

**Review by tw15egan — APPROVED** [review]
LGTM! 👍 ✅

File: src/components/ComponentIndexPage/ComponentIndexListItem.js (hunk 1 above)
Comment (emyarod) [reply src/components/ComponentIndexPage/ComponentIndexListItem.js:65]: ok I see, strange that it was not formatted before for whatever reason

**Review by emyarod — APPROVED** [review]

**Pull request merged**